GitUI/Editor/FileViewerInternal.cs

```diff
@@ -201,8 +205,60 @@ public async Task FindNextAsync(bool searchForwardOrOpenWithDifftool)
 
         public event EventHandler HScrollPositionChanged;
         public event EventHandler VScrollPositionChanged;
+        public event EventHandler TopScrollReached;
+        public event EventHandler BottomScrollReached;
         public new event EventHandler TextChanged;
 
+        private void TextArea_MouseWheel(object sender, MouseEventArgs e)
+        {
+            var isScrollingTowardTop = e.Delta > 0;
+            var isScrollingTowardBottom = e.Delta < 0;
+
+            if (isScrollingTowardTop && (TextEditor.ActiveTextAreaControl.VScrollBar.Value == 0))
+            {
+                CallTopScrollReachedEvent(sender, e);
+            }
+
+            if (isScrollingTowardBottom && (!TextEditor.ActiveTextAreaControl.VScrollBar.Visible
+                                            || TextEditor.ActiveTextAreaControl.VScrollBar.Value + TextEditor.ActiveTextAreaControl.VScrollBar.Height >
+                                            TextEditor.ActiveTextAreaControl.VScrollBar.Maximum))
```

Comment:
Keep on the same line, it is the same expression
```suggestion
                                            || TextEditor.ActiveTextAreaControl.VScrollBar.Value + TextEditor.ActiveTextAreaControl.VScrollBar.Height > TextEditor.ActiveTextAreaControl.VScrollBar.Maximum))
```

If you find it too long, create a variable on line:216 and reuse it across the method:
```cs
var vScrollBar = TextEditor.ActiveTextAreaControl.VScrollBar;
```

Reply:
have to call it scrollBar though to get Hungarian notation error...